### ISSUE TITLE
Allow a recursive option on rmdir

### ISSUE THREAD

**Issue by amxmln** [isomorphic-git/lightning-fs#71]
The `rmdir` function in newer versions of Node allows for a `recursive` option that when set to `true` makes the command behave more like a `rm -rf`, which I think can be very useful in many situations (such as deleting a cloned but no longer needed git repo for example).

The documentation of `lightning-fs` suggests that its `rmdir` also takes an options object, but I browsing through the source I couldn’t find anything that seemed to be using that object (I might have missed something though :sweat_smile:). I feel like adding a native `recursive` option could really benefit the library and perhaps would even allow for a more efficient way than a recursive "rimraf" implemented on top of the library, since if I understood the structure of `lightning-fs` correctly it should be possible to just gather a list of all inodes contained in the directory, delete them and then delete the directory itself in a single transaction, right?

Sorry if that suggested solution is useless, but I hope the rest of my point still stands.

Keep up the great work! :tada:

**Comment by SaadBazaz** [comment]
Needed!

**Comment by jcubic** [comment]
@SaadBazaz do you want to contribute?

**Comment by SaadBazaz** [comment]
@jcubic I can try giving it a go. Any pointers to where I can start?

**Comment by jcubic** [comment]
You should just get familiar with the code, it was a while when I read the code I don't remember where you should start. But there are not much code in this project.

**Comment by kaytwo** [comment]
FWIW I ran into this issue and just used `unlink` on a directory and it worked as I expected it to (equivalent to `rm -rf`).

**Comment by amxmln** [comment]
Yes, it will not actually delete the files, they will just become inaccessible, because their parent is removed. They will still take up disk space, however, like you said.

**Comment by 7flash** [comment]
mkdir also missing recursive option

**Comment by tom-sze** [comment]
export async function clearDirectory(dir) {
    // same as rm -r
    for (let item of await window.pfs.readdir(dir)) {
        const item_path = `${dir}/` + item;
        if ((await window.pfs.stat(item_path)).type === 'file') {
            await window.pfs.unlink(item_path);
        } else {
            await clearDirectory(item_path);
            await window.pfs.rmdir(item_path);
        }
    }
}
